src/registries/autofill_rules.ts

```diff
@@ -77,6 +86,53 @@ function calculateIncrementBasedOnGroup(group: number[]) {
   return increment;
 }
 
+function getEqualInterval(intervals: DateIncrement[], key: keyof DateIncrement) {
+  if (intervals.length < 2) {
+    return intervals[0]?.[key] || 0;
+  }
+
+  const equal = intervals.every((interval) => interval[key] === intervals[0][key]);
+  return equal ? intervals[0][key] : undefined;
+}
+
+/**
+ * Get the step for a group of dates
+ * if the increment is the same everywhere (+1 day, +1 month, +1 year)
+ * return the same increment
+ * else if 2 dates, return the difference
+ * else return 0
+ * */
+function calculateDateIncrementBasedOnGroup(group: number[]) {
+  if (group.length < 2) {
+    return 1;
+  }
+  const jsDates = group.map((date) => toJsDate(date, DEFAULT_LOCALE));
+
+  const datesIntervals = getDateIntervals(jsDates);
+
+  const daysInterval = getEqualInterval(datesIntervals, "days");
+  const monthsInterval = getEqualInterval(datesIntervals, "months");
+  const yearsInterval = getEqualInterval(datesIntervals, "years");
+
+  // ## algo un peu pété il faut quand meme fixer les deltas
+  // et véfier que c'est les memes tout simplement ?
```

Comment:
🥖

Reply:
The algo seems _fine_, it doesn't do the same thing as Gsheet/Excel if we have an increment of 1 day and 1 year for example, but I don't really understand what they do 🤷